src/generateSdk/generateSdkApi.ts

```diff
@@ -74,6 +87,30 @@ export async function sdkGeneratorApiHandler(
     };
 }
 
+function checkInvalidHttpMethods(sdkClass: SdkClassConfiguration, astClass: ClassDefinition): void {
+    const methods: MethodDefinition[] = [];
+    let methodsString = "";
+    for (const method of sdkClass.methods) {
+        if (method.type === TriggerType.http) {
+            const astMethod = astClass.methods.find((m) => m.name === method.name);
+            if (
+                astMethod &&
+                (astMethod.params.length != 1 ||
+                    astMethod.params[0].paramType.type != AstNodeType.RequestType ||
+                    astMethod.params[0].optional != false ||
+                    astMethod.returnType.type != AstNodeType.PromiseType ||
```

Comment:
I would also add the possibility for the return type to be directly `ResponseType`, to make it generic for other languages that will never return promises.